src/collections/maple_tree/mod.rs

```diff
@@ -29,6 +30,13 @@ impl<T> M<T> {
             M::None => panic!(),
         }
     }
+
+    fn clone(self) -> M<T> {
```

Comment:
I think you don't need to implement clone like this. Either #[derive(Clone)] on struct M, or impl Clone for M<T> (inside of impl M is not correct).